upgraded code and replaced reprecated switch

TheRojam opened the pull request:
_done two things_ - first one is ansible and the second yunohost related
I upgraded this playbook to version min. 2.10, which introduced collection naming. 
I also removed the deprecated switch for mail (--mail/-m) and replaced it w/ one for domain (--domain/-d).

TheRojam committed:
upgraded code for ansible version 2.10/3

TheRojam committed:
updated meta data

TheRojam committed:
Merge pull request #1 from TheRojam/upgrade/ansible_version_3

Upgrade/ansible version 3

TheRojam committed:
switch -m (mail) was deprecated in favor of -d (--domaim)
also added example users w/ {{ domain }}

TheRojam committed:
Merge pull request #2 from TheRojam/error/mail2domain

switch -m (mail) was deprecated in favor of -d (--domaim)

TheRojam committed:
added steps to secure Authentication w/o Password

TheRojam committed:
Merge pull request #3 from TheRojam/add/admin_ssh_user

added steps to secure Authentication w/o Password

TheRojam committed:
corrected ansible errors

sylvainar commented:
Thanks for your PR. I'd be happy to merge, however I'm not confortable with the fact that we disable password connection, it'll break existing setups. You should either remove it or add a parameter to the role so it's disabled by default.

Is the role still working if you're not providing any ssh key for admins? I haven't use yunohost in a while and I don't know if it's mandatory.

TheRojam commented:
there are some difficulties :D but its not about the admin users. the sshkeys are not mandatory, buts for vps or cloud servers this is best practice or big suggestion from the view of security. 👍🏿

TheRojam committed:
corrected domain 2 mail_domain

TheRojam committed:
corrected parameter for replace-module from regex 2 regexp

TheRojam committed:
added condition for usage of now nopassword

cchaudier committed:
ci: add linter and syntax check from project gci-tpl

cchaudier committed:
Merge branch '1-remove-admin-users-feature' into 'master'

✅Resolve "remove admin users feature"

Closes #1

See merge request lydra/yunohost/ansible-yunohost!2

artybdrlt committed:
refactor: removed action related to admin_users task

This feature is system-related, not yunohost-related

cchaudier committed:
fix: no playbook syntax chech because this repo as no playbook

cchaudier committed:
Merge branch '4-ci-add-ansible-lint-and-checks' into 'master'

Resolve "ci: add ansible lint and checks"

Closes #4

See merge request lydra/yunohost/ansible-yunohost!1

artybdrlt committed:
refactor: changed vars to reflect new variable names

refactor: changed variables pattern to be more consistent with ansible good practises and more flexible to use.
refactor: remove useless comment
doc: Readme in English and French

cchaudier committed:
Merge branch '2-refactor-vars' into 'master'

✅Resolve "refactor vars"

Closes #2 and #9

See merge request lydra/yunohost/ansible-yunohost!3

artybdrlt committed:
♻refactor: Command ansible module across the repo for better security

Shell module is not needed here.

According to various sources (https://www.youtube.com/watch?v=57gAqKvAKck or https://stackoverflow.com/questions/56663332/difference-between-shell-and-command-in-ansible) it is not useful to use shell ansible module when not working with operands. Therefore I have decided to switch every actions to command module, more secure. Ansible-lint says "Shell should only be used when piping, redirecting or chaining commands"

cchaudier committed:
Merge branch '10-use-command-module' into 'master'

✅Resolve "Use command module"

Closes #10

See merge request lydra/yunohost/ansible-yunohost!6

artybdrlt committed:
♻refactor: using latest ansible module to be more future-proof and support latest ansible versions

Ansible-lint:
```
Include has some unintuitive behaviours depending on if it is running in a static or dynamic in play or in playbook context, in an effort to clarify behaviours we are moving to a new set modules (ansible.builtin.include_tasks, ansible.builtin.include_role, ansible.builtin.import_playbook, ansible.builtin.import_tasks) that have well established and clear behaviours.
This module will still be supported for some time but we are looking at deprecating it in the near future.
```

Fix: I have decided to go for ansible include_tasks module because it is more versatile and on a par with this module (for more info about differences between new modules, see [here](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_module.html)

cchaudier committed:
Merge branch '14-ansible-include-module-is-to-be-deprecated' into 'master'

✅Resolve "ansible include module is to be deprecated"

Closes #14

See merge request lydra/yunohost/ansible-yunohost!7

cchaudier committed:
Merge branch '12-ansible-test-when-is-smth-rather-than-smth' into 'master'

✅Resolve "fix: ci lint"

Closes #12

See merge request lydra/yunohost/ansible-yunohost!5

artybdrlt committed:
fix: ansible-lint errors

- fix: [201] Trailing whitespace REMOVED
- fix: meta-no-tags: Use 'galaxy_tags' rather than 'categories' REPLACED + role in Debian 10 only
- fix: [601] Don't compare to literal True/False FIXED

Commit:
refactor: Better understanding at how the role interacts with ynh

artybdrlt committed:
fix: default GitHub branch is main, needed to pull code from ansible-galaxy + lydra is author of this role

cchaudier committed:
Merge branch '23-github-default-branch' into 'main'

✅Resolve "GitHub default branch"

Closes #23

See merge request lydra/yunohost/ansible-yunohost!11

cchaudier committed:
Merge branch '18-add-badges-on-readme-for-github' into 'main'

Resolve "Add badges on readme for github"

Closes #18

See merge request lydra/yunohost/ansible-yunohost!10

cchaudier committed:
doc: add some awesome badges

cchaudier committed:
doc: change ansible-galaxy id

cchaudier committed:
Merge branch '24-doc-change-ansible-id' into 'main'

Resolve "doc: change ansible id"

Closes #24

See merge request lydra/yunohost/ansible-yunohost!12

cchaudier committed:
Merge branch '8-doc-review' into 'main'

Resolve "doc: review"

Closes #8

See merge request lydra/yunohost/ansible-yunohost!8

cchaudier committed:
Merge branch '15-doc-licence-information' into 'main'

✅Resolve "doc: licence information"

Closes #15

See merge request lydra/yunohost/ansible-yunohost!9

artybdrlt committed:
refactor: Added GNU licence info + lydra copyright

Commit:
fix: defaults folder for Ansible well-being

cchaudier committed:
Merge branch '30-default-folder-instead-of-defaults' into 'main'

✅Resolve "default folder instead of defaultS"

Closes #30

See merge request lydra/yunohost/ansible-yunohost!13

artybdrlt committed:
feat(yunohost): add SMTP relay support

cchaudier committed:
Merge branch '31-add-smtp-relay' into 'main'

✅Resolve "add smtp relay"

Closes #31

See merge request lydra/yunohost/ansible-yunohost!14

cchaudier committed:
Merge branch '32-feat-add-app-post_install' into 'main'

Resolve "feat: add app post_install"

Closes #32

See merge request lydra/yunohost/ansible-yunohost!16

artybdrlt committed:
feat: add app post_install

TheRojam committed:
Merge pull request #4 from LydraFr/main

merge improvements from fork

TheRojam commented:
i already merge some improvements by @cchaudier

cchaudier commented:
Thanks @TheRojam for merged our code.
